### PR TITLE
add firefox suggest events to clients_daily

### PR DESCRIPTION
**Pull request by rebecca-burwei**
I would like to add the last recorded event per-day per-client for [this firefox suggest event](https://firefox-source-docs.mozilla.org/browser/urlbar/firefox-suggest-telemetry.html#contextservices-quicksuggest-opt-in-dialog) to clients_daily

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql [hunk 1]
```diff
@@ -1,21 +1,56 @@
...
 FROM
-  telemetry.events
-WHERE
-  submission_date = @submission_date
-GROUP BY
-  submission_date,
-  sample_id,
-  client_id
+  telemetry_events
+LEFT JOIN
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql:53]: This LEFT JOIN will end up being quite expensive given the number of unique client IDs. I believe we can express this logic within the existing single aggregation like this:

```
  ARRAY_AGG(
    IF(event_category = 'contextservices.quicksuggest' AND event_method = 'opt_in_dialog', 
      event_object, 
      NULL) 
    IGNORE NULLS 
    ORDER BY session_start_time DESC
  )[OFFSET(0)] AS firefoxsuggest_optin,
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_joined_v1/schema.yaml [hunk 2]
```diff
@@ -320,6 +320,9 @@ fields:
   mode: REPEATED
   name: experiments
   type: RECORD
+- mode: NULLABLE
+  name: firefoxsuggest_optin
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_joined_v1/schema.yaml:324]: I feel like this name should be a bit more verbose; just calling it `optin` sounds like a boolean to me. How about `firefoxsuggest_opt_in_dialog_choice`?

**Review by jklukas — CHANGES_REQUESTED** [review]
Left some questions about naming and performance. If we can reach agreement in threads, I'd be happy to actually push the modifications, as I'll need to make some other changes before getting this merged per https://mozilla.github.io/bigquery-etl/cookbooks/common_workflows/#add-a-new-field-to-clients_daily

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql (hunk 1 above)
Comment (rebecca-burwei) [reply sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql]: sounds good

**Comment by rebecca-burwei** [comment]
> Left some questions about naming and performance. If we can reach agreement in threads, I'd be happy to actually push the modifications, as I'll need to make some other changes before getting this merged per https://mozilla.github.io/bigquery-etl/cookbooks/common_workflows/#add-a-new-field-to-clients_daily

Sounds good - thanks for your help. I made those changes. Should I try to follow the instructions in the cookbook?

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql [hunk 3]
```diff
@@ -3,19 +3,25 @@ SELECT
...
+  ARRAY_AGG(
+  IF
+    (event_category = 'contextservices.quicksuggest'
+      AND event_method = 'opt_in_dialog',
+      event_object,
+      NULL) IGNORE NULLS
+  ORDER BY
+    session_start_time DESC )[
+OFFSET
+  (0)] AS firefoxsuggest_optin,
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql:19]:
```suggestion
  (0)] AS firefoxsuggest_opt_in_dialog_choice,
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql [hunk 4]
```diff
@@ -3,19 +3,25 @@ SELECT
...
   telemetry.events
 WHERE
   submission_date = @submission_date
 GROUP BY
-  submission_date,
-  sample_id,
-  client_id
+  1,
+  2,
+  3
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_event_v1/query.sql:27]:
```suggestion
  submission_date,
  sample_id,
  client_id
```

Our preferred style is to write out column names explicitly in group by and sort by. I am failing to find this practice documented, though.

**Review by jklukas — COMMENTED** [review]
> Should I try to follow the instructions in the cookbook?

Go for it! I'll need to run the final step of deploying schema change after the PR is ready for merge, but you should be able to prep the schema updates. But feel free to ask for help if you run into issues with the tooling. There are sometimes gaps that are hard to debug.

**Comment by jklukas** [comment]
> should I create a GCP service account

Actually creating a service account isn't necessary. See https://docs.telemetry.mozilla.org/cookbooks/bigquery/access.html#api-access and in particular `gcloud auth application-default login` which basically creates temporary service account creds from your user creds.

**Comment by rebecca-burwei** [comment]
I found out the data I'm looking to add for clients daily is already there. :D

**Comment by jklukas** [comment]
An epic conclusion. Such drama! 🍿

**Comment by rebecca-burwei** [comment]
Hahaha! @jklukas in case you're interested, this was the last step I got to. 
```
bash-3.2$ ./bqetl query schema update telemetry_derived.clients_daily_event_v1 --update_downstream
Error: No module named 'jnius'
```

I did run into some other import issues along the way that I resolved by updating my bqetl CLI, but I wasn't sure here where `jnius` needed to be installed, if in a virtual env or something else.

**Comment by jklukas** [comment]
Yeah, `jnius` is a Java interop package. Step 4 of https://github.com/mozilla/bigquery-etl#installing-bqetl has you use `java-dependencies.txt` which pulls in `jnius`. But then `jnius` needs to be able to find a Java environment that it likes, which would probably have been the next hurdle.

**Pull request closed without merging**